docs/guides/web/nginx-mainline.md

```diff
@@ -0,0 +1,282 @@
+
+First, make sure your machine is updated:
+
+```bash
+sudo dnf update
+```
+
+Now, make sure the *dnf-utils* package is installed, and any command-line text editor you may want:
+
+```bash
+sudo dnf install dnf-utils
+```
+
+Once that’s all installed, power up your text editor of choice. You’ll want to create a file called (for the sake of simplicity) “nginx.repo”, and put it in `/etc/yum.repos.d/`. You can do this real quick like so:
+
+```bash
+sudo nano /etc/yum.repos.d/nginx.repo
+```
+
+In that file, paste this bit of code, unmodified:
+
+```bash
+[nginx-stable]
+name=nginx stable repo
+baseurl=http://nginx.org/packages/centos/$releasever/$basearch/
+gpgcheck=1
+enabled=1
+gpgkey=https://nginx.org/keys/nginx_signing.key
+module_hotfixes=true
+
+[nginx-mainline]
+name=nginx mainline repo
+baseurl=http://nginx.org/packages/mainline/centos/$releasever/$basearch/
+gpgcheck=1
+enabled=0
+gpgkey=https://nginx.org/keys/nginx_signing.key
+module_hotfixes=true
+```
+
+This code basically just lets you use the *Nginx*-made-and-hosted repositories for CentOS, and it allows you use the previously-mentioned “stable” branch if you want to. I mean, don’t. But you could.
+
+Save the file with Control-S (if using *nano*) and exit with Control-X.
+
+## Installing and Running Nginx
+
+Now, let’s enable the repository file you just made with one simple command:
+
+```bash
+sudo yum-config-manager --enable nginx-mainline
+```
+
+Then, install *Nginx*:
+
+```bash
+sudo dnf install nginx
+```
+
+The terminal will ask you if you’re fine with installing the repository’s GPG key. You need that, so choose “Y” for yes.
+
+Once the installation is done, start *Nginx* and enable it to automatically start on reboot all in one go with:
+
+```bash
+sudo systemctl enable --now nginx
+```
+
+To verify that the lastest version of *Nginx* has been installed, run:
+
+```bash
+nginx -v
+```
+
+From there, you could just start dropping HTML files into the `/usr/share/nginx/html/` directory to build a simple, static website. The configuration file for the default website/virtual host is called “default.conf” and it’s in `/etc/nginx/conf.d/`.
+
+## Configuring the Firewall
+
+!!! Note
+
+    If you are installing Nginx on a container such as LXD/LXC or Docker, you can just skip this part for now. The firewall should be handled by the host OS.
+
+If you try to view a web page at your machine’s IP address or domain name from another computer, you’re probably going to get a big fat nothing. Well, that’ll be the case as long as you have a firewall up and running.
+
+Here’s how to open up the necessary ports to actually see your web pages with *firewalld*, Rocky Linux’s default firewall with the `firewall-cmd` command. To add a new port, just run this:
+
+```bash
+sudo firewall-cmd --permanent --zone=public --add-port=80/tcp
```

Comment:
Why not using the service name? `firewall-cmd --add-service=http`
It's easier to understant.
```suggestion
sudo firewall-cmd --permanent --zone=public --add-service=http
```

Reply:
For the service suggestion, I think I'll mention both. To be fair to me, I lifted this section from my upcoming guide on LXD web servers, and I'm pretty sure the ports matter there.

Reply:
I think this is a rule that should be set at the editorial charter level. I propose that when the service is standard, we should use the name of the service, when the service is custom, we should use the port.

Reply:
Neither of you are wrong here. They are both (port or service) fundamentally the same. I want someone to show me how I can fine-tune firewalls down to a specific IP address allow rule. If they can show me this, and it's not a complete cluster, then I'll actually move away from iptables to firewalld.